### PR TITLE
Add bash installation script for macos or linux users.

### LINKED ISSUE #482: [Feature] Bash install script
## Feature

I propose that we release a bash script in order to simplify the installation process of the kuberbuilder binary

Commands for macos
```
export version=1.0.5
export arch=amd64
curl -L -O https://github.com/kubernetes-sigs/kubebuilder/releases/download/v${version}/kubebuilder_${version}_darwin_${arch}.tar.gz
tar -zxvf kubebuilder_${version}_darwin_${arch}.tar.gz
sudo mv kubebuilder_${version}_darwin_${arch} /usr/local/kubebuilder
```

to be converted to something like this 

```
curl -L https://git.io/getLatestKubebuilder | sh -
#!/bin/sh
#
# This file will be fetched as: curl -L https://git.io/getLatestKubebuilder | sh -
# so it should be pure bourne shell, not bash (and not reference other scripts)
#
# The script fetches the latest kubebuilder release candidate and untars it.
# It lets users to do curl -L https://git.io/getLatestIstio | KUBEBUILDER_VERSION=0.1.5 sh -
# for instance to change the version fetched.

OS="$(uname)"
if [ "x${OS}" = "xDarwin" ] ; then
  OSEXT="darwin"
else
  OSEXT="linux"
fi
ARCH=amd64

if [ "x${KUBEBUILDER_VERSION}" = "x" ] ; then
  KUBEBUILDER_VERSION=$(curl -L -s https://api.github.com/repos/kubernetes-sigs/kubebuilder/releases/latest | \
                  grep tag_name | sed "s/ *\"tag_name\": *\"\\(.*\\)\",*/\\1/")
fi

# https://github.com/kubernetes-sigs/kubebuilder/releases/download/v1.0.5/kubebuilder_1.0.5_darwin_amd64.tar.gz

KUBEBUILDER_VERSION=${KUBEBUILDER_VERSION#"v"}
NAME="kubebuilder_${KUBEBUILDER_VERSION}"
URL="https://github.com/kubernetes-sigs/kubebuilder/releases/download/v${KUBEBUILDER_VERSION}/${NAME}_${OSEXT}_${ARCH}.tar.gz"
echo "Downloading $NAME from $URL ..."
curl -L "$URL" | tar xz
# TODO: change this so the version is in the tgz/directory name (users trying multiple versions)
echo "Downloaded into $NAME: "
ls "${NAME}_${OSEXT}_${ARCH}/bin"
sudo mv ${NAME}_${OSEXT}_${ARCH} /usr/local/kubebuilder
echo "Add kubebuilder to your path; e.g copy paste in your shell and/or ~/.profile:"
echo "export PATH=$PATH:/usr/local/kubebuilder/bin"
```

### PR DESCRIPTION
**Pull request by cmoulliard**
- Add bash installation script for macos or linux users. 
- download . latest version of the project or specific version, untar it and move project under /usr/local/bin/kubebuilder
- Fix #482

### CODE REVIEW

**Comment by droot** [comment]
@cmoulliard Its blocked on CLA.

**Comment by cmoulliard** [comment]
I tried many times to get an email's validation from "linuxfoundation" to create an account for the ICLA since yesterday without success.

File: scripts/install.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+KUBEBUILDER_VERSION=${KUBEBUILDER_VERSION#"v"}
+NAME="kubebuilder_${KUBEBUILDER_VERSION}"
+URL="https://github.com/kubernetes-sigs/kubebuilder/releases/download/v${KUBEBUILDER_VERSION}/${NAME}_${OSEXT}_${ARCH}.tar.gz"
+echo "Downloading $NAME from $URL ..."
+curl -L "$URL" | tar xz
+
+echo "Downloaded these executable files into $NAME: "
+ls "${NAME}_${OSEXT}_${ARCH}/bin"
+sudo mv ${NAME}_${OSEXT}_${ARCH} /usr/local/kubebuilder
```
Comment (hidekazuna) [inline scripts/install.sh]: This line has changed to the following line in #470.
mv kubebuilder_${version}_darwin_${arch} kubebuilder && sudo mv kubebuilder /usr/local/

Comment (cmoulliard) [reply scripts/install.sh]: If `/usr/local/kubebuilder` folder already exists, then we can't use `mv kubebuilder /usr/local` directly but a `sudo rm -rf /usr/local/kubebuilder` will be first needed. Make sense @hidekazuna ?

Comment (hidekazuna) [reply scripts/install.sh]: If /usr/local/kubebuilder directory already exists, it is better to warn it and stop executing than force installing.
If force Installing, kubebuilder may be upgraded or downgraded unintentionally.

Comment (cmoulliard) [reply scripts/install.sh]: I agree with you. We will then exit in this case and add a warning message to the user that prior to upgrade, in this case, they have to first delete the existing folder or backup it.

**Comment by cmoulliard** [comment]
As my PR contains 2 non needed commits, I will close this one submit a new PR if you agree including also the modifications requested for the bash script -> https://github.com/cmoulliard/kubebuilder/commit/b85612e657c9ab0b56506b085a0e1c89dc76f0ec

**Comment by cmoulliard** [comment]
New PR will be submitted

**Pull request closed without merging**